custom_components/ipx800v4/binary_sensor.py

```diff
@@ -36,26 +40,50 @@ async def async_setup_entry(
 
     for device in devices:
         if device.get(CONF_TYPE) == TYPE_VIRTUALOUT:
-            entities.append(VirtualOutBinarySensor(device, controller, coordinator))
+            on_value = 0 if device.get(CONF_INVERT) else 1
+            entities.append(VirtualOutBinarySensor(device, controller, coordinator, on_value))
         elif device.get(CONF_TYPE) == TYPE_DIGITALIN:
-            entities.append(DigitalInBinarySensor(device, controller, coordinator))
+            on_value = 0 if device.get(CONF_INVERT) else 1
+            entities.append(DigitalInBinarySensor(device, controller, coordinator, on_value))
 
     async_add_entities(entities, True)
 
 
 class VirtualOutBinarySensor(IpxEntity, BinarySensorEntity):
     """Representation of a IPX Virtual Out."""
 
+    def __init__(
+        self,
+        device_config: dict,
+        ipx: IPX800,
+        coordinator: DataUpdateCoordinator,
+        on_value: int
+    ) -> None:
+        """Initialize the VirtualOutBinarySensor."""
+        super().__init__(device_config, ipx, coordinator)
+        self.on_value = on_value
+
```

Comment:
as we now set the Inverted value to False for all config entities, you can defined `self._inverted_value = device_config[CONF_INVERT]` directly in the class IpxEntity

Reply:
yeah, make sense. I guess that's the reason why my first attempt wasn't working